app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -23,67 +23,93 @@ import androidx.navigation.Navigation
 import kotlinx.android.synthetic.main.fragment_history.view.*
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.Dispatchers.Main
+import kotlinx.coroutines.MainScope
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 import mozilla.components.concept.storage.VisitType
+import mozilla.components.lib.state.Store
+import mozilla.components.lib.state.ext.observe
 import mozilla.components.support.base.feature.BackHandler
 import org.mozilla.fenix.BrowserDirection
 import org.mozilla.fenix.BrowsingModeManager
-import org.mozilla.fenix.FenixViewModelProvider
 import org.mozilla.fenix.HomeActivity
 import org.mozilla.fenix.R
 import org.mozilla.fenix.components.Components
+import org.mozilla.fenix.components.StateViewModel
 import org.mozilla.fenix.ext.components
 import org.mozilla.fenix.ext.getHostFromUrl
 import org.mozilla.fenix.ext.nav
 import org.mozilla.fenix.ext.requireComponents
-import org.mozilla.fenix.mvi.ActionBusFactory
-import org.mozilla.fenix.mvi.getAutoDisposeObservable
-import org.mozilla.fenix.mvi.getManagedEmitter
 import org.mozilla.fenix.share.ShareTab
 import java.util.concurrent.TimeUnit
 
 @SuppressWarnings("TooManyFunctions")
 class HistoryFragment : Fragment(), BackHandler {
-
-    private lateinit var historyComponent: HistoryComponent
-    private val navigation by lazy { Navigation.findNavController(requireView()) }
+    private lateinit var historyStore: HistoryStore
+    private lateinit var historyView: HistoryView
+    private lateinit var historyInteractor: HistoryInteractor
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? = inflater
-        .inflate(R.layout.fragment_history, container, false).also { view ->
-            historyComponent = HistoryComponent(
-                view.history_layout,
-                ActionBusFactory.get(this),
-                FenixViewModelProvider.create(
-                    this,
-                    HistoryViewModel::class.java,
-                    HistoryViewModel.Companion::create
-                )
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_history, container, false)
+        val viewModel = StateViewModel.get(
+            this,
+            HistoryState(
+                items = listOf(), mode = HistoryState.Mode.Normal
             )
+        )
+        historyStore = Store(
+            viewModel.state,
+            ::historyStateReducer
+        )
+        historyStore.observe(this) {
+            viewModel.update(it)
```

Comment:
We just pushed up a PR so we can remove this line: https://github.com/mozilla-mobile/fenix/pull/4060 :)

Basically, we use the store directly so we don't have to update it. This also solve the problem that we may be updating it here on a different thread (not the main thread, as the lib-state reducer has a dedicated one).